test/e2e/autoscale.go

```diff
@@ -60,8 +62,9 @@ const (
 // TestContext includes context for autoscaler testing.
 type TestContext struct {
 	t                 *testing.T
+	logf              logging.FormatLogger
```

Comment:
Break out this refactoring? Seems valid even at HEAD :thinking:

Reply:
This is tightly coupled with the other changes in this PR. If I don't pass logf and use T.Infof directly the upgrade tests end with RACE errors because the Golang testing framework uses that concurrently with the function called from upgrade tests. I guess this could be sent in a separate commit in this PR but don't see a good reason for sending this in a separate PR.

Reply:
Right, but it can easily be frontloaded by making all the changes necessary on HEAD and by referencing the issue you're working on, stating exactly what you just said as the reason for the PR. It's even better than today as there' no explanation on why this is needed currently.